lib/modules/manager/npm/update/dependency/index.ts

```diff
@@ -1,9 +1,12 @@
 import { dequal } from 'dequal';
 import type { PackageJson } from 'type-fest';
 import { logger } from '../../../../../logger';
+import { regEx } from '../../../../../util/regex';
 import { matchAt, replaceAt } from '../../../../../util/string';
 import type { UpdateDependencyConfig } from '../../../types';
 
+const patchReg = regEx('patch:.*@(npm:)?.*#.*');
```

Comment:
I think we should ignore dep if it has any unsupported protocol.

Reply:
These are the `npm` protocols that we might face :
``` NPM("npm"),
    GIT("git"),
    GIT_SSH("git+ssh"),
    GIT_HTTP("git+http"),
    GIT_HTTPS("git+https"),
    GIT_FILE("git+file"),
    GITHUB("github"),
    FILE("file"),
    LINK("link"),
    PATCH("patch"),
    PORTAL("portal"),
    EXEC("exec"),
    HTTP("http"),
    HTTPS("https");```

the normal versioning is the `npm` protocol, in yarn.lock of yarn v2.0+  it will show {name}@npm:{version}
you might not see it in package.json directly since its redundant and no one puts it in there but it is a thing
try it out:
add to the package.json dependencies:
```
  "dependencies": {
    "numpy": "npm:0.0.1"
  },
```
it works, its possible.
i believe you are familiar with the `git` protocols, since its commonly used in some projects..
but other stuff is a bit rare to find.

i'm not sure how to support all these protocols FULLY currently, they are not commonly used, but i believe we will face them eventually, since yarn 2.0+ uses them normally.

Reply:
we don't need to support them yet, but we should notice and ignore them to not propose wrong updates

Reply:
well i'm not sure what we do support, so i dont want to go breaking anything,
this needs lots of testing for each protocol to make sure we dont support it. 
unless any of you guys @rarkins / @viceice can tell what we dont support for sure, so i can ignore the rest,
Do we have any documentation for what protocols we support?

Reply:
i think we don't support any other protocol, so we should ignore them. @rarkins WDYT?

Reply:
Ignore

Reply:
are you saying you do not support:
![image](https://user-images.githubusercontent.com/42116482/157006728-e71b2902-252e-4f0f-9708-2d98bc26226f.png)

Reply:
i will ignore them, but it might break some stuff you already have working, since this is the replace strings functionality of npm

Reply:
there you go, 
I broke it, you already support git protocols
![image](https://user-images.githubusercontent.com/42116482/157065313-9a3a04be-2e9f-45bf-b674-d9447b107255.png)

Reply:
ok we have another problem with another protocol:
ok i have another suggestion, 
![image](https://user-images.githubusercontent.com/42116482/157650376-f30f8cc3-c8a7-45d0-b9c6-c2b954be4027.png)


we also support file protocol,

i suggest not to ignore any protocol, as we might be supporting them while we are not aware of it. can i remove the code?